bolt-connection/src/connection/connection-channel.js

```diff
@@ -198,6 +198,25 @@ export default class ChannelConnection extends Connection {
             if (!this.databaseId) {
               this.databaseId = dbConnectionId
             }
+
+            if (metadata.hints) {
+              const receiveTimeoutRaw =
+                metadata.hints['connection.recv_timeout_seconds']
+              if (
+                receiveTimeoutRaw !== null &&
+                receiveTimeoutRaw !== undefined
+              ) {
+                const receiveTimeoutInSeconds = toNumber(receiveTimeoutRaw)
+                if (receiveTimeoutInSeconds > 0) {
```

Comment:
Not sure if even feasible in JS's limited number type, but we also don't expect floats from the server.

Reply:
I will create tests with float number to limit this option. One question, should `3.0` be considered an invalid value?

Reply:
I don't know how the JS drivers handles numbers internally. But the server is expected to sent an Integer (strongly typed through packstream). It might well be, however, that at this point in the code, the type information is already lost. In that case, I'd say 3.0 is also ok.

Reply:
At that point, I already lost this information. Since 3 is the same as 3.0 for JS. So, I'm verifying if it's a Integer using the JS Number definition of integer.